pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -431,6 +431,24 @@ CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.label_unnest(label_array anyarray)
 AS $function$ SELECT unnest(label_array) $function$;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.label_unnest(anyarray) to prom_reader;
 
+-- safe_approximate_row_count returns the approximate row count of a hypertable if timescaledb is installed
+-- else returns the approximate row count in the normal table. This prevents errors in approximate count calculation
+-- if timescaledb is not installed, which is the case in plain postgres support.
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.safe_approximate_row_count(table_name_input REGCLASS) RETURNS BIGINT
+    LANGUAGE PLPGSQL
+AS
+$$
+BEGIN
+    IF SCHEMA_CATALOG.is_timescaledb_installed() THEN
```

Comment:
```suggestion
    IF SCHEMA_CATALOG.is_timescaledb_installed()  AND (SELECT count(*) > 0 from _timescaledb_catalog.hypertable WHERE format('%I.%I', schema_name, table_name)::regclass=table_name_input) THEN
```

Reply:
Note the two changes:
1. We put both the conditions in one IF, simplifying into one IF/ELSE instead of IF/IF/ElSE/ELSE
2. instead of casting the regclass to text, we cast the text to regclass and we use format. Format makes sure we avoid any possible sql injection.

Reply:
> cast the text to regclass and we use format

This is great :+1: